src/repo.rs

```diff
@@ -1188,15 +1222,13 @@ impl Repo {
                 // Handle doc changes: sync the document.
                 let local_repo_id = self.get_repo_id().clone();
                 if let Some(info) = self.documents.get_mut(&doc_id) {
-                    if !info.note_changes() {
-                        // Stop here if the document wasn't actually changed.
-                        return;
+                    if info.note_changes() {
+                        self.documents_with_changes.push(doc_id.clone());
```

Comment:
I think a comment is in order here: "only run the `documents_with_changes` workflow if there was a change, but always generate potential sync messages(below)"